caproto/sync/client.py

```diff
@@ -542,10 +542,16 @@ def _write(chan, data, metadata, timeout, data_type, notify):
     logger.debug("Detected native data_type %r.", chan.native_data_type)
     # abundance of caution
     ntype = field_types['native'][chan.native_data_type]
-    if data_type is None:
-        # Handle ENUM: If data is INT, carry on. If data is STRING,
-        # write it specifically as STRING data_Type.
-        if (ntype is ChannelType.ENUM) and isinstance(data[0], bytes):
+    if (data_type is None) and (ntype is ChannelType.ENUM):
+        # Change data_type to STRING if data contains string-like data, or
+        # iterable of string-like data
+        stringy_data = False
+        if isinstance(data, (str, bytes)):
+            stringy_data = True
+        if hasattr(data, '__getitem__') and isinstance(data[0], (str, bytes)):
```

Comment:
`data = tuple()` I believe would be problematic. Should add condition for `len(data)` as well

Reply:
I hadn't thought of that at all, good catch.